src/config/etc/sssd.api.d/sssd-ipa.conf

```diff
@@ -1,10 +1,13 @@
+# Format:
+# option = type, subtype, mandatory[, default]
 [provider/ipa]
 ipa_domain = str, None, false
 ipa_server = str, None, false
 ipa_backup_server = str, None, false
 ipa_hostname = str, None, false
 ipa_deskprofile_search_base = str, None, false
 ipa_subid_ranges_search_base = str, None, false
+ipa_access_order = str, None, false,
```

Comment:
You have to remove the `,` at the end as well because with the `,` it means that the default is an empty string.

Reply:
I wonder why this comma is there. I never meant to put it.

But then I wonder if I shouldn't add "" as default value

Reply:
Hi,

no, the default should be empty here. The `*.conf` files in src/config/etc are used by SSSD's python package `SSSDConfig` which can be used to create and modify `sssd.conf` from python. The default value is used when creating a new configuration or configuration section to make sure that a somewhat working configuration is created so that SSSD can start and hopefully handle some basic requests.

I think the main purpose for the default here was the `services` option, where the default is `nss, pam`, i.e. the nss and pam responders will be started by default. Since the SSSD main code does not set a default for this option it is important that this option is set in sssd.conf. I only found one other default value for `homedir_substring`.

In the case of `ipa_access_order` a default value here is not needed because if this option is not set the default from `ipa_basic_opts` will be used, so an explicit entry in sssd.conf is not needed.

HTH

bye,
Sumit

Reply:
Thanks for the explanation. I'll just remove the comma.